subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -113,6 +111,8 @@ static inline bool ctrl_is_unexpected(struct ll_conn *conn, uint8_t opcode);
 #endif /* CONFIG_BT_CTLR_LE_ENC */
 
 #if defined(CONFIG_BT_CTLR_CONN_PARAM_REQ)
+static inline void cpr_active_check_and_reset(struct ll_conn *conn);
+static inline void cpr_active_reset(void);
```

Comment:
Is there any good reason why these are `inline`? The compiler should be able to optimize it well based on the optimization parameter. 

See also https://www.kernel.org/doc/html/v4.10/process/coding-style.html#the-inline-disease

The comment applies to all the new functions added.

Reply:
These functions operate on global variable (compile time constant), do nothing more that assignments. Since, the functions are called more than a single place, if compile optimizes for space (which is the default in zephyr), then there will be icache miss leading to wait states which is in addition to branching operation.

Reply:
Interesting. Is this something you have seen/measured before? 

Either way, if that is the case, could I bother you to add that as a comment somewhere in the file, so that it is clear, and so that no one else attempts to remove the `inline` for these functions?

Reply:
Yes, seen and measured some of the inline function use in the controller. If I recollect, for example `adv_pdu_adva_get` in ull_adv.c has two callers and without `inline` compiler branches to the function.